lib/Common/Memory/Recycler.h

```diff
@@ -979,6 +978,10 @@ class Recycler
     PageAllocator backgroundProfilerPageAllocator;
     DListBase<ArenaAllocator> backgroundProfilerArena;
 #endif
+
+    // destruct autoHeap after backgroundProfilerPageAllocator;
+    HeapInfo autoHeap;
```

Comment:
backgroundProfilerArena allocated heap block from autoHeap. If autoHeap destructed before backgroundProfilerArena, the destructor for the latter will run on some freed heap blocks.

---
In reply to: [66281817](https://github.com/Microsoft/ChakraCore/pull/1096#discussion_r66281817) [](ancestors = 66281817)